lighthouse-core/test/gather/driver-test.js

```diff
@@ -535,4 +543,75 @@ describe('Multiple tab check', () => {
       });
     });
   });
+
+  describe('Security check', () => {
+    it('does not reject when page is secure', async () => {
+      const secureSecurityState = {
+        explanations: [],
+        securityState: 'secure',
+      };
+      driverStub.on = driverStub.once = createOnceStub({
```

Comment:
this poor `driverStub` was never meant to be used repeatedly :)

Can you also break the cycle and make a `new Driver(connection)` within each of these tests? We can deal with having a fresh `connection` later

Reply:
FWIW I'm also rather fond of keeping all the mocking close to the test like I did over in manifest PR with `jest.spyOn`. Given the way the past few reviews have gone I'm guessing @brendankenny feels the opposite though so take it with a grain of salt 😂

Reply:
> Given the way the past few reviews have gone I'm guessing @brendankenny feels the opposite

Ha, I'm not entirely sure how to interpret that. I also like maximizing locality so you can see what's actually been done to the mock you have to use. But I am generally against mocks if at all possible, if that's what that means :)

Reply:
> I also like maximizing locality

phew! :) 

> Ha, I'm not entirely sure how to interpret that.

only a joke no worries, just I feel like I've lead @Hoten chasing too many wild geese lately 😆

Reply:
> only a joke no worries

lol I was trying to read too deeply :) :)